### ISSUE TITLE
Filter read IDs based on an input file

### ISSUE THREAD

**Issue by mwhamgenomics** [EdinburghGenomics/Fastq-Filterer#12]
We want to be able to supply an input file containing read IDs to be filtered from each Fastq:

```
@instrument:run:flowcell:lane:1101:x:y 1:0:0:0
@instrument:run:flowcell:lane:1102:x:y 1:0:0:0
@instrument:run:flowcell:lane:2102:x:y 2:0:0:0
```

We should also redirect filtered reads to a file (but not trimmed reads).
CR840

**Comment by tcezard** [comment]
The file provided will have the read names without read1/read2 information because the read will have to be removed from both fastq file.
Something along the line of
```
@read_01
@read_04
```

**Comment by mwhamgenomics** [comment]
That's a bit of a problem, because the read number is buried inside the header:

`@instrument:run:flowcell:lane:tile:x:y read_number:filter_flag:0:sample_number`

The only way that would work for matching both reads is if we do regexes, or only match up to the space in the middle of the line (kind of similar to the tile filtering, which splits by `:` and walks along to the tile ID).

**Comment by tcezard** [comment]
I would split at the space and take the first part 
It is unique and is propagated in the bam files which means any alignment based method will be able to return it.

**Comment by mwhamgenomics** [comment]
Closed in #13
